src/NuGet.Core/NuGet.Commands/PackCommandRunner.cs

```diff
@@ -516,8 +531,14 @@ private PackageArchiveReader BuildFromProjectFile(string path)
                 factory.GetProjectProperties()[property.Key] = property.Value;
             }
 
+            NuGetVersion version = null;
+            if (_packArgs.Version != null)
+            {
+                version = new NuGetVersion(_packArgs.Version);
+            }
+
             // Create a builder for the main package as well as the sources/symbols package
-            PackageBuilder mainPackageBuilder = factory.CreateBuilder(_packArgs.BasePath);
+            PackageBuilder mainPackageBuilder = factory.CreateBuilder(_packArgs.BasePath, version, _packArgs.Suffix, buildIfNeeded: true);
```

Comment:
Darn, but OK if that's what's needed.

---
In reply to: [63772517](https://github.com/NuGet/NuGet.Client/pull/606#discussion_r63772517) [](ancestors = 63772517)